app/helpers/search_helper.rb

```diff
@@ -9,7 +9,7 @@ def escape_method_name(method_name)
   end
 
   def method_anchor(method)
-    "method-#{method.instance_method? ? "i" : "c"}-#{escape_method_name(method.name)}"
+    "method-#{method.method_type[0]}-#{escape_method_name(method.name)}"
```

Comment:
This is kind of experimental. For some theoretical cases (in the future) we should change it to `[0..1]` or something smarter.

Reply:
It's pretty important to the RDoc code that this is only `'i'` or `'c'`, as these anchors are what RDoc makes as well. I don't know that anything other than `'i'` or `'c'` is possible as `method_type` goes, but I just wanted to make sure that you're aware of that.

Reply:
> but I just wanted to make sure that you're aware of that.

I didn't know about this, thank you.

> This is kind of experimental. For some theoretical cases (in the future) we should change it to `[0..1]` or something smarter.

And, I guess, then this (original) behavior can be broken. I'll think how to make it better (more explicit and correct).

Reply:
You know… I think, the current code (from this PR) is most correct for such RDoc behavior. Or maybe somebody has better suggestions?

Reply:
Here's how RDoc does it (and what I believe Colby mimicked):

https://github.com/ruby/rdoc/blob/4592938ce0622523fd811b547ea5525db0ecb251/lib/rdoc/method_attr.rb#L216-L220

Reply:
> Here's how RDoc does it (and what I believe Colby mimicked):
> 
> https://github.com/ruby/rdoc/blob/4592938ce0622523fd811b547ea5525db0ecb251/lib/rdoc/method_attr.rb#L216-L220

Oh… thank you. I've reverted these changes.